benchmark/benchmark.py

```diff
@@ -60,6 +63,53 @@ def write_metadata(metadata: Dict[str, Any]) -> None:
         log.error("Failed to write metadata", exc_info=True)
 
 
+def upload_results_to_s3(bucket_name: str, region: str) -> None:
+    """
+    Upload benchmark results to S3 bucket using the AWS CLI.
+    Only uploads results from multirun directories.
+    """
+    try:
+        hydra_config = HydraConfig.get()
+
+        if hydra_config.mode == RunMode.MULTIRUN:
+            output_path = Path(hydra_config.runtime.output_dir)
+
+            source_path = output_path.parent
+
+            parts = output_path.parts
+
+            if len(parts) >= 3:
+                date_part, time_part = parts[-3:-1]
```

Comment:
We should make this part more readable. On idea that was discussed: break up the path into its parts, find the latest multirun occurence and get the next two parts. However, if we can assert that output_path.parts, we can just unpack it:
```
 date_part, time_part, _ = parts
```